client/app/components/packages/pas-form/edit.js

```diff
@@ -64,19 +68,29 @@ export default class PasFormComponent extends Component {
     return this.submittableChanges.isValid;
   }
 
-  @action
-  async save() {
-    await this.saveableChanges.save();
-
-    this.args.onSave();
+  get isSavingOrSubmitting() {
+    return this.save.isRunning || this.submit.isRunning;
   }
 
-  @action
-  async submit() {
-    await this.submittableChanges.save();
+  @task(function* () {
+    yield this.saveableChanges.execute();
 
-    this.args.onSubmit();
-  }
+    yield this.args.onSave();
+
+    yield this.saveableChanges.rollback();
+  })
+  save;
+
+  @task(function* () {
```

Comment:
I'm sad they haven't fixed the syntax for this.

You used to be able to say this:

```
@task()
*submit () {
}
```

Are you sure this isn't allowed? I think it looks a little less awkward.

Reply:
Oh, I didn't come access/think of this. I can give it a go!

Reply:
I'm getting an error when I try that syntax:
![image](https://user-images.githubusercontent.com/3311663/81835834-b2ccc100-9510-11ea-9079-e17fda2ead07.png)